projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -241,12 +270,19 @@ function ( $payment_gateway ) {
 				$properties = $this->process_event_properties(
 					'woocommerceanalytics_product_checkout',
 					array(
-						'pq'               => $cart_item['quantity'],
-						'payment_options'  => $enabled_payment_options,
-						'device'           => wp_is_mobile() ? 'mobile' : 'desktop',
-						'guest_checkout'   => 'Yes' === $guest_checkout ? 'Yes' : 'No',
-						'create_account'   => 'Yes' === $create_account ? 'Yes' : 'No',
-						'express_checkout' => 'null',
+						'pq'                => $cart_item['quantity'],
+						'payment_options'   => $enabled_payment_options,
+						'device'            => wp_is_mobile() ? 'mobile' : 'desktop',
+						'guest_checkout'    => 'Yes' === $guest_checkout ? 'Yes' : 'No',
+						'create_account'    => 'Yes' === $create_account ? 'Yes' : 'No',
+						'express_checkout'  => 'null',
+						'shipping_option'   => $this->get_shipping_option_for_item( $cart_item_key ),
+						'products_count'    => count( wc()->cart->get_cart() ),
+						'coupon_used'       => count( wc()->cart->applied_coupons ) > 0 ? '1' : '0',
```

Comment:
Do we need to account for cases when the result is not countable?

Reply:
Thanks! I've done this, also added checks for the order value in d8ac52ee6c7bac6d01891860dcd9f353a0bd1435